### ISSUE TITLE
collections of Literal constructed from tuple of Literal

### ISSUE THREAD

**Issue by crusaderky** [python/mypy#7533]
mypy 0.720, python 3.7.4, typing_extensions 3.7.4

mypy outputs a false positive when creating a collection of Literal using a constructor that takes as an argument a Tuple of Literal:

(inlined mypy output for readibility)
```python
from typing import Set, FrozenSet, List, Tuple
from typing_extensions import Literal

T = Literal[1, 2, 3]
l: List[T] = [1, 2]
t1: Tuple[T, ...] = (1, 2)
t2: Tuple[T, T] = (1, 2)
s1: Set[T] = {1, 2}
s2: Set[T] = set([1, 2])
s3: Set[int] = set((1, 2))
f1: FrozenSet[T] = frozenset([1, 2])
f2: FrozenSet[int] = frozenset((1, 2))
# no error on any of the above

s4: Set[T] = set((1, 2))
# error: Argument 1 to "set" has incompatible type "Tuple[int, int]"; expected "Iterable[Union[Literal[1], Literal[2], Literal[3]]]"

f3: FrozenSet[T] = frozenset((1, 2))
# error: Argument 1 to "frozenset" has incompatible type "Tuple[int, int]"; expected "Iterable[Union[Literal[1], Literal[2], Literal[3]]]"
```

**Comment by JukkaL** [comment]
It looks like that when inferring the type of a tuple expression, mypy doesn't use the item type from an `Iterable[...]` context as the type context. This isn't specific to literal types either.

**Comment by JukkaL** [comment]
Added literal types as the topic since it seems like this will most likely happen with literal types.

**Comment by GoldsteinE** [comment]
Possible workaround:
```python
T = Literal[1, 2, 3]
x: Tuple[T, ...] = (1, 2)
s4: Set[T] = set(x)
```

**Comment by erictraut** [comment]
This appears to have been fixed at some point. The latest version of mypy no longer emits any errors, and the revealed types in the example look correct.
